ruby_event_store/lib/ruby_event_store/client.rb

```diff
@@ -46,35 +46,27 @@ def delete_stream(stream_name)
     end
 
     def read_events_forward(stream_name, start: :head, count: @page_size)
-      raise IncorrectStreamData if stream_name.nil? || stream_name.empty?
-      page = Page.new(@repository, start, count)
-      deserialized_events(@repository.read_events_forward(stream_name, page.start, page.count))
+      deserialized_events(read.stream(stream_name).limit(count).from(start).each)
```

Comment:
I did not expect this... I thought we would

* Step 1
  * keep using old methods in repository such as read_events_forward
  * delegate read_events_forward in repo to read(specification)
  * deprecate read_events_forward
* Step 2
  * start using repo.read
  * remove read_events_forward from repo

but that doesn't make sense as we would be using deprecated methods internally so we would need to detect if repo support old or new way. Hmmm. Probably not worth the hassle